datalad/distribution/get.py

```diff
@@ -264,8 +264,15 @@ def _get_flexible_source_candidates_for_submodule(ds, sm):
             # template - this can happen if a pure Git repository is queried
             # for a dataset id, for example.
             lgr.debug('Caught a key error in the generation of a submodule '
-                      'URL using the template %s (%s)', tmpl, exc_str(e))
-            misbuilt_url[name] = (tmpl, e)
+                      'URL using the template %s (%s). The URL will not be '
+                      'constructed.', tmpl, exc_str(e))
+            from datalad.support import ansi_colors
+            msg = ansi_colors.color_word(
```

Comment:
@bpoldrack : unsure if it is good practice to do the ansi colors at this stage rather than at result rendering. Could you let me know?

Reply:
In my view, it should be done at rendering. Here we don't know yet (or shouldn't) whether there even is a terminal output, etc.
So, I'd just have the renderer rendering this particular field in that way, while field itself only contains text.

Reply:
ok, thx